src/PkgServer.jl

```diff
@@ -90,13 +90,28 @@ function start(;kwargs...)
     update_registries()
 
     @sync begin
-        global last_registry_update
-        global registry_task = @spawn while true
-            last_registry_update = now()
-            sleep(1)
-            @try_printerror begin
-                forget_failures()
-                update_registries()
+        global last_registry_update = now()
+        global registry_update_task = @spawn begin
+            while true
+                last_registry_update = now()
+                sleep(1)
+                @try_printerror begin
+                    forget_failures()
+                    update_registries()
+                end
+            end
+        end
+
+        # Registry watchdog; if `last_registry_update` doesn't change
+        # for more than 20 minutes, we kill ourselves so that we can restart
+        global registry_watchdog_task = @spawn begin
+            while istaskstarted(registry_update_task)
```

Comment:
What is this check doing? Is the case of a dead update task handled elsewhere?

Reply:
Right; if I ever gracefully stop the registry update task, I want this to gracefully stop as well.  I think actually this can be made better by just doing a `while true` and then `break`'ing out if `istaskdone` and restarting immediately if `istaskfailed`.